src/soplex.hpp

```diff
@@ -3860,6 +3860,114 @@ bool SoPlexBase<R>::getPrimalRational(VectorBase<Rational>& vector)
       return false;
 }
 
+template <class R>
+bool SoPlexBase<R>::getRowPrimalValue(int i, R& value) {
+  if(_realLP != 0 && hasSol())
+  {
+    _syncRealSolution();
+    value = _realLP->computePrimalActivity(i, _solReal._primal, true);
+    return true;
+  }
+  else
+    return false;
+}
+
+template <class R>
+bool SoPlexBase<R>::getRowPrimalValueRational(int i, Rational& value) {
+  if(_rationalLP != 0 && hasSol())
+  {
+    _syncRationalSolution();
+    value = _rationalLP->computePrimalActivity(i, _solRational._primal);
+    return true;
+  }
+  else
+    return false;
+}
+
+template <class R>
+bool SoPlexBase<R>::getRowsPrimalValue(const std::set<int>& indices, VectorBase<R>& vector) {
+  if(_realLP != 0 && hasSol())
+  {
+    assert(vector.dim() >= static_cast<int>(indices.size()));
+    _syncRealSolution();
+    _realLP->computePrimalActivity(indices, _solReal._primal, vector, true);
+    return true;
+  }
+  else
+    return false;
+}
+
+
+template <class R>
+bool SoPlexBase<R>::getRowsPrimalValueReal(const std::set<int>& indices, R* p_vector, int i) {
+  if(_realLP != 0 && hasSol())
+  {
+    assert(i >= static_cast<int>(indices.size()));
```

Comment:
does the `i` even make sense here? Why not just use `indices.size()`?

Reply:
Since `i` is really the size of the `R` `p_vector`, this assertion ensures that there are at least `indices.size()` slots in `p_vector`. Changing `i` to `dim` will help with clarity.
This said, `dim` should really be `>= nRows()` instead of `indices.size()`, unless you want to deal with sparse vectors in this context.